### PR TITLE
Random search

### PR DESCRIPTION
**Pull request by aaronkl**
Implements a simple random search baseline as described in #206 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: emukit/examples/gp_bayesian_optimization/random_search.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        self.rd = RandomDesign(parameter_space=parameter_space)
+
+    def compute_next_points(self, loop_state: LoopState, context: dict=None) -> np.ndarray:
+        """
+        :param loop_state: Object that contains current state of the loop
+        :param context: Contains variables to fix through optimization of acquisition function. The dictionary key is
+                        the parameter name and the value is the value to fix the parameter to.
+        :return: (n_points x n_dims) array of next inputs to evaluate the function at
+        """
+        return self.rd.get_samples(1)
```
Comment (marpulli) [inline emukit/examples/gp_bayesian_optimization/random_search.py]: This currently ignores the context. You could do:

```python
if context_manager is not None:
    sample = context_manager.context_free_space.sample_uniform(1)
    sample = context_manager.expand_vector(sample)
else:
    sample = self.parameter_space.sample_uniform(1)
return sample
```

Comment (aaronkl) [reply emukit/examples/gp_bayesian_optimization/random_search.py]: good point, I will add that

File: emukit/examples/gp_bayesian_optimization/random_search.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+        if x_init is not None and y_init is not None:
+            loop_state = create_loop_state(x_init, y_init, cost_init)
+        else:
+            loop_state = None
+
+        super().__init__(candidate_point_calculator, model_updaters, loop_state=loop_state)
+
+    def get_results(self):
+        return BayesianOptimizationResults(self.loop_state)
```
Comment (apaleyes) [inline emukit/examples/gp_bayesian_optimization/random_search.py]: That's the only weird thing in the whole PR. RandomSearch has no reason to return BayesOpt results. What if I want to use it as a baseline for quadrature?

File: integration_tests/emukit/bayesian_optimization/test_random_search.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import numpy as np
```
Comment (apaleyes) [inline integration_tests/emukit/bayesian_optimization/test_random_search.py]: Given what I said above, this file may need some splitting/refactoring

File: emukit/examples/gp_bayesian_optimization/random_search.py (hunk 2 above)
Comment (aaronkl) [reply emukit/examples/gp_bayesian_optimization/random_search.py]: The idea was to provide the same output as BayesOpt for comparisons. However, if we move to the core library I will drop it.

**Comment by aaronkl** [comment]
I like the idea of merging it into the core library, since I could also see some other cases where these functionalities might be useful

File: emukit/core/loop/model_updaters.py [hunk 4]
```diff
@@ -24,6 +24,16 @@ def update(self, loop_state: LoopState) -> None:
         pass
 
 
+class NoopModelUpdate(ModelUpdater):
```
Comment (apaleyes) [inline emukit/core/loop/model_updaters.py]:
```suggestion
class NoopModelUpdater(ModelUpdater):
```

File: emukit/core/loop/model_updaters.py [hunk 5]
```diff
@@ -24,6 +24,16 @@ def update(self, loop_state: LoopState) -> None:
...
+    def update(self, loop_state: LoopState) -> None:
+        """
+        Dummy model update, can be used for example for random search
```
Comment (apaleyes) [inline emukit/core/loop/model_updaters.py]:
```suggestion
        Dummy model updater that does nothing, can be used for example for random search
```

File: emukit/benchmarking/loop_benchmarking/random_search.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+
+from emukit.core.loop.outer_loop import OuterLoop
```
Comment (apaleyes) [inline emukit/benchmarking/loop_benchmarking/random_search.py]: Inside the library all imports should be relative

File: tests/emukit/benchmarking/test_random_search.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+
+def test_random_search_with_init_data():
+    np.random.seed(42)
+    branin_fcn, parameter_space = branin_function()
+
+    x_init = parameter_space.sample_uniform(5)
+    y_init = branin_fcn(x_init)
+    cost_init = np.ones([5, 1])
+    print(cost_init.ndim)
```
Comment (apaleyes) [inline tests/emukit/benchmarking/test_random_search.py]: Remove print, i suppose some debugging leftover?

**Pull request merged**